### PR TITLE
Add Hannah65-RGB PCB Rev.1

### PR DESCRIPTION
**Pull request by mechlovin**
## Description

Add Hannah65-RGB PCB rev.1
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by drashna — APPROVED** [review]

**Comment by drashna** [comment]
Because this moves around different keyboard folders and their locations, please target develop.  You may need to rebase this after you've done so.

File: keyboards/mechlovin/hannah65/rgb_led/config.h [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#pragma once
```
Comment (tzarc) [inline keyboards/mechlovin/hannah65/rgb_led/config.h:1]: Please add a copyright header here.

File: keyboards/mechlovin/hannah65/rgb_led/rev1/config.h [hunk 2]
```diff
@@ -0,0 +1,31 @@
+#pragma once
```
Comment (tzarc) [inline keyboards/mechlovin/hannah65/rgb_led/rev1/config.h:1]: Please add a copyright header here.

File: keyboards/mechlovin/hannah65/standard_led/config.h [hunk 3]
```diff
@@ -0,0 +1,8 @@
+#pragma once
```
Comment (tzarc) [inline keyboards/mechlovin/hannah65/standard_led/config.h:1]: Please add a copyright header here.

File: keyboards/mechlovin/hannah65/standard_led/mechlovin9/config.h [hunk 4]
```diff
@@ -0,0 +1,5 @@
+#pragma once
```
Comment (tzarc) [inline keyboards/mechlovin/hannah65/standard_led/mechlovin9/config.h:1]: Please add a copyright header here.

File: keyboards/mechlovin/hannah65/rgb_led/rules.mk [hunk 5]
```diff
@@ -0,0 +1,3 @@
+
+
```
Comment (tzarc) [inline keyboards/mechlovin/hannah65/rgb_led/rules.mk:2]:
```suggestion
```

File: keyboards/mechlovin/hannah65/standard_led/rev1/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT_all(
+        KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_DEL,  KC_HOME,
+        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,          KC_PGUP,
+        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_ENT,                    KC_PGDN,
+        KC_LSFT, KC_NUBS, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_LSFT,          KC_UP,   KC_END,
+        KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                             KC_RALT, KC_RCTL,          KC_LEFT, KC_DOWN, KC_RGHT
+    ),
+
+
```
Comment (tzarc) [inline keyboards/mechlovin/hannah65/standard_led/rev1/keymaps/default/keymap.c:27]:
```suggestion
```

File: keyboards/mechlovin/hannah65/standard_led/rev1/keymaps/via/keymap.c [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    ),
+    [3] = LAYOUT_all(
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,                   KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS,                            KC_TRNS,                            KC_TRNS, KC_TRNS,          KC_TRNS, KC_TRNS, KC_TRNS
+    ),
+
+
```
Comment (tzarc) [inline keyboards/mechlovin/hannah65/standard_led/rev1/keymaps/via/keymap.c:48]:
```suggestion
```

**Review by tzarc — CHANGES_REQUESTED** [review]
Just some minor stuff to do with the PR checklist.

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Looks like you have a merge conflict here.

**Comment by mechlovin** [comment]
I don't see conflict here, can you tell me?
I will correct it. Thank you

**Comment by fauxpark** [comment]
image

**Comment by mechlovin** [comment]
@fauxpark @drashna hi guy, is it ok?
I don't have much experience using github. Thanks for your help

**Comment by drashna** [comment]
This should cover how to update stuff: https://docs.qmk.fm/#/newbs_git_resolving_merge_conflicts

**Comment by drashna** [comment]
If possible, any folder that requires the `DEFAULT_FOLDER` to compile should have the `.noci` file in it, to prevent the CI from trying to compile the same keyboard revision multiple times.

**Comment by mechlovin** [comment]
> If possible, any folder that requires the `DEFAULT_FOLDER` to compile should have the `.noci` file in it, to prevent the CI from trying to compile the same keyboard revision multiple times.


Thank you, this is new to me.
I thought my computer had a problem when I first saw the .noci file :D

**Comment by drashna** [comment]
The `.noci` is ***very*** new.  It was added a week or so ago, actually, and it is to prevent duplicate folders from being compiled.  This way, we can drastically speed up the Travis CI runs, so they stop timing out due to running too long.  And it has definitely helped!

**Comment by drashna** [comment]
Lint isn't happy, not sure if that's critical, @yanfali @Erovia ?

**Comment by noroadsleft** [comment]
@drashna That lint check fails because not all of the revision folders have readme files. I believe the lint check expects a `readme.md` file in every keyboard folder that has a `rules.mk`. Not a deal-breaker IMO; the link check should probably be updated, rather than the codebase.

File: keyboards/mechlovin/hannah65/rgb_led/rev1/rules.mk [hunk 8]
```diff
@@ -0,0 +1,3 @@
+RGB_MATRIX_ENABLE = IS31FL3731
+RGBLIGHT_ENABLE = yes
+DEFAULT_FOLDER = mechlovin/hannah65/rgb_led/rev1
```
Comment (noroadsleft) [inline keyboards/mechlovin/hannah65/rgb_led/rev1/rules.mk:3]: This line isn't needed.

```suggestion
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]

**Comment by mechlovin** [comment]
PCB has faulty part in hardware design, RGB matrix is ​​faulty due to not complying with datasheet, some red LEDs blink when using RGB_MATRIX_KEYPRESSES and RGB_MATRIX_KEYRELEASES.
Q9, page 6. 
https://github.com/UltimateHackingKeyboard/datasheets/blob/master/leds/IS31FL3731%20Application%20Note%20Rev.C.pdf

**Comment by mechlovin** [comment]
I will close this PR and open another PR, use another LED driver.
Thank you for support!

**Pull request closed without merging**